apollo-router/src/plugins/record_replay/replay.rs

```diff
@@ -0,0 +1,418 @@
+use std::borrow::Cow;
+use std::collections::HashSet;
+use std::ops::ControlFlow;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use console::style;
+use http::Method;
+use http::Uri;
+use multimap::MultiMap;
+use tokio::fs;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt as TowerServiceExt;
+
+use super::recording::Recording;
+use crate::context::Context;
+use crate::layers::ServiceBuilderExt;
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::services::execution;
+use crate::services::router;
+use crate::services::subgraph;
+use crate::services::supergraph;
+use crate::services::TryIntoHeaderName;
+use crate::services::TryIntoHeaderValue;
+
+#[derive(Debug)]
+pub(crate) struct Replay {
+    recording: Recording,
+    pub(crate) report: Arc<Mutex<Vec<ReplayReport>>>,
+}
+
+#[allow(dead_code)]
+impl Replay {
+    pub(crate) async fn from_file(recording_file: String) -> Result<Self, BoxError> {
+        let recording = fs::read_to_string(recording_file).await?;
+        let recording: Recording = serde_json::from_str(&recording)?;
+        Ok(Self::new(recording))
+    }
+
+    pub(crate) fn new(recording: Recording) -> Self {
+        Self {
+            recording,
+            report: Arc::from(Mutex::from(vec![])),
```

Comment:
NIT: I think it should work
```suggestion
            report: Arc::default(),
```